lua/lint/linters/resharper.lua

```diff
@@ -0,0 +1,60 @@
+local function find_upward(extension)
+  local dir = vim.fn.getcwd()
+  while dir and dir ~= "/" do
+    local files = vim.fn.globpath(dir, "*" .. extension, false, true)
+    if not vim.tbl_isempty(files) then
+      return files[1]
+    end
+    dir = vim.fn.fnamemodify(dir, ":h")
+  end
+  return nil
+end
+
+return function()
+  local root_file = find_upward(".sln")
+  if not root_file then
+    root_file = find_upward(".csproj")
+  end
+
+  if not root_file then
+    vim.notify("No solution file or csproj found in parent directories", vim.log.levels.WARN)
+    return {}
+  end
+
+  local current_buffer = vim.api.nvim_get_current_buf()
+  local filepath = vim.api.nvim_buf_get_name(current_buffer)
+
+  if filepath == "" then
+    vim.notify("Current buffer has no file associated", vim.log.levels.WARN)
+    return {}
+  end
+
+  local solution_directory = vim.fn.fnamemodify(root_file, ":h")
+
+  local relative_filepath = filepath:sub(#solution_directory + 2)
+  local cache_directory = vim.fn.stdpath("cache") .. "/nvim-lint-resharper"
```

Comment:
Could this be changed to only search `.sln` or `.csproj` relative to the current dir?
Adding root dir heuristics as part of the linter is out of scope for nvim-dap. `cwd` can be parameterized as part of the `try_lint` call for users who have `cwd != project_dir`.

Reply:
> So I guess it's fine as long the user always has csproj or sln in the cwd. For me, that rarely is the case because 

You could still do something as described in https://github.com/mfussenegger/nvim-lint/issues/482#issuecomment-1999185606

And at some point in the future there might be a solution based on https://github.com/neovim/neovim/issues/34622 

So for now I'd restrict resharper to only search downward from cwd